internal/definition/awsintegration/schema_integration.go

```diff
@@ -34,9 +34,16 @@ func newIntegrationSchema() map[string]*schema.Schema {
 			Description: "Whether the integration is enabled or not",
 		},
 		"auth_method": {
-			Type:        schema.TypeString,
-			Computed:    true,
-			Description: "The mechanism used to authenticate with AWS. Use one of `signalfx_aws_external_integration` or `signalfx_aws_token_integration` to define this",
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice(
+				[]string{
+					string(integration.EXTERNAL_ID),
+					string(integration.SECURITY_TOKEN),
+				},
+				false,
+			),
+			Description: "The mechanism used to authenticate with AWS. Must be either `ExternalId` or `SecurityToken`.",
```

Comment:
Is this really necessary? Can't we leave the field spec as-is? 

Otherwise, if I understand correctly, all TF users would need to update their TF scripts?

Reply:
Yeah, so I had made this change to make it explicit on what the auth provider is that is being used. 

This is mostly to address the original comment on using auth method as the means of figuring out what settings to use. This would be required in the create case since this field is currently computed so a user can not currently set it; it would require the most recent commit to be reverted.